src/main/java/com/zendesk/maxwell/schema/ddl/TableCreate.java

```diff
@@ -28,44 +30,48 @@ public TableCreate (String database, String table, boolean ifNotExists) {
 	}
 
 	@Override
-	public Schema apply(Schema originalSchema) throws SchemaSyncError {
-		Schema newSchema = originalSchema.copy();
+	public ResolvedTableCreate resolve(Schema schema) throws SchemaSyncError {
+		Database d = schema.findDatabaseOrThrow(this.database);
 
-		Database d = newSchema.findDatabase(this.database);
-		if ( d == null )
-			throw new SchemaSyncError("Couldn't find database " + this.database);
+		if ( ifNotExists && d.hasTable(table) )
+			return null;
 
+		Table table = null;
 		if ( likeDB != null && likeTable != null ) {
-			applyCreateLike(newSchema, d);
+			table = resolveLikeTable(schema);
 		} else {
-			Table existingTable = d.findTable(this.table);
-			if (existingTable != null) {
-				if (ifNotExists) {
-					return originalSchema;
-				} else {
-					throw new SchemaSyncError("Unexpectedly asked to create existing table " + this.table);
-				}
-			}
-			Table t = d.buildTable(this.table, this.charset, this.columns, this.pks);
-			t.setDefaultColumnCharsets();
+			table = new Table(this.database, this.table, this.charset, this.columns, this.pks);
+			resolveCharsets(d.getCharset(), table);
 		}
 
-		return newSchema;
+		if ( schema.getCaseSensitivity() == CaseSensitivity.CONVERT_TO_LOWER )
+			table.name = table.name.toLowerCase();
+
+		return new ResolvedTableCreate(table);
 	}
 
-	private void applyCreateLike(Schema newSchema, Database d) throws SchemaSyncError {
-		Database sourceDB = newSchema.findDatabase(likeDB);
+	private Table resolveLikeTable(Schema schema) throws SchemaSyncError {
+		Database sourceDB = schema.findDatabaseOrThrow(likeDB);
+		Table sourceTable = sourceDB.findTableOrThrow(likeTable);
 
-		if ( sourceDB == null )
-			throw new SchemaSyncError("Couldn't find database " + likeDB);
+		Table copiedTable = sourceTable.copy();
+		copiedTable.database = this.database;
+		copiedTable.name = this.table;
 
-		Table sourceTable = sourceDB.findTable(likeTable);
-		if ( sourceTable == null )
-			throw new SchemaSyncError("Couldn't find table " + likeDB + "." + likeTable);
+		return copiedTable;
+	}
 
-		Table t = sourceTable.copy();
-		t.rename(this.table);
-		d.addTable(t);
+	private void resolveCharsets(String dbCharset, Table resolved) {
+		if ( this.charset != null )
```

Comment:
err, this isn't true.  it's true before, when we called `database.buildTable()`, but not now.  I think there's probably another pass of coding in here to make the resolver-code smart and the data-code dumb -- ie the resolver should be telling the `Table` "you have this character set, based on the nature of the CREATE TABLE and what I know about the database", the `Table` itself shouldn't know about implicit charsets and the like